backend/src/main/java/com/google/models/Alert.java

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.models;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.models.Anomaly;
+import com.google.models.Timestamp;
+import java.util.List;
+import java.util.ArrayList;
+import java.time.format.DateTimeParseException;
+
+/** Store alert-related data. */
+public final class Alert {
+  public static final String ALERT_ENTITY_KIND = "alert";
+  public static final String STATUS_PROPERTY = "status";
+  public static final String ANOMALIES_LIST_PROPERTY = "anomaliesList";
+  public static final String RESOLVED_STATUS = "resolved";
+  public static final String UNRESOLVED_STATUS = "unresolved";
+
+  private final Timestamp timestampDate;
+  private final List<Anomaly> anomalies;
+  private String status;
```

Comment:
how many different alert statuses will we have in the future? would it make sense to have this an `enum` or should we keep it as string?

Reply:
currently, we only have two statuses in mind: `resolved` and `unresolved`, but there could be more later on. 
i think alert statuses would be good as `enum`, but if they're `enum`, would they be defined within the `Alert` class like this?
``` 
public final class Alert {
  public static final enum StatusType {
    resolved,
    unresolved
  }

  // Rest of code...
}
```
This would allow methods that create an `Alert` object to access the available status types. But then, they will constantly have to type `Alert.StatusType.resolved` or `Alert.StatusType.unresolved` when creating an `Alert`. (1)

Or we can still take in `String status` and assign them to a `StatusType` enum by checking `String status`, but then that could defeat the point of having status types as `enum`. (2)

I guess another way would be to define the `StatusType` enum outside the `Alert` class but since `StatusType` is only associated with an `Alert` I don't think it should be outside the `Alert` class. (3)

Personally, I think (1) is the best, but I could be totally wrong lol...

Reply:
From my personal perspective:
- since we already know the set of these statuses is limited (currently limited to two) and known during the compile-stage, enums are reasonable.
- strings might be a better option if the set of statuses is open/variable and not known when compiling

I saw you already created the `// TODO: Check whether status is valid` since strings can allow any type of input for the status which will be not needed given the limited scope if using enums.

**if** we decide to go with the enum statuses, i agree with you that it should probably be inside the `Alert` class, since they're referring to alert statuses. You can imagine other classes might end up having other types of `statuses` which will go in their corresponding class.

You mentioned the following trade-off:

> But then, they will constantly have to type Alert.StatusType.resolved or Alert.StatusType.unresolved when creating an Alert. 

which I agree, that this will be a trade-off where the engineer will have to write a bit more to get the status in expense of just Alert.RESOLVED_STATUS.

Reply:
thanks for sharing your insights and perspective!

i think i will go with using `enum` within the `Alert` class as the statuses will be limited to a few and will be known during compile-stage. i think the trade-off should be manageable.